packages/language-server/src/xml-view-diagnostics.ts

```diff
@@ -0,0 +1,108 @@
+import { map, drop } from "lodash";
+import { assertNever } from "assert-never";
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+  DiagnosticTag
+} from "vscode-languageserver-types";
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { DocumentCstNode, parse } from "@xml-tools/parser";
+import { buildAst } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import {
+  UI5XMLViewIssue,
+  validateXMLView,
+  XMLViewIssueSeverity
+} from "@ui5-language-assistant/xml-views-validation";
+
+export function getXMLViewDiagnostics(opts: {
+  document: TextDocument;
+  ui5Model: UI5SemanticModel;
+}): Diagnostic[] {
+  const documentText = opts.document.getText();
+  const { cst, tokenVector } = parse(documentText);
+  const xmlDocAst = buildAst(cst as DocumentCstNode, tokenVector);
+  const issues = validateXMLView({
+    xmlView: xmlDocAst,
+    model: opts.ui5Model
+  });
+  const diagnostics = validationIssuesToLspDiagnostics(issues, opts.document);
+  return diagnostics;
+}
+
+function validationIssuesToLspDiagnostics(
+  issues: UI5XMLViewIssue[],
+  document: TextDocument
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = map(issues, currIssue => {
+    const commonDiagnosticPros: Diagnostic = {
+      range: {
+        start: document.positionAt(currIssue.offsetRanges[0].start),
+        // Chevrotain's end offsets are none inclusive
+        end: document.positionAt(currIssue.offsetRanges[0].end + 1)
+      },
+      severity: toLspSeverity(currIssue.severity),
+      source: "UI5 Language Assistant",
+      message: currIssue.message
+    };
+
+    // TODO: I think we should ditch the related issues for now, it does not work well for the only use case I can think of currently (deprecated on ending tag)
```

Comment:
What do you think on this comment @tal-sapan

Reply:
What do you mean doesn't work well? What does it do?

Reply:
Related issues seem to just be implemented a link to other ranges.
so it does not replace actually having multiple distinct separate issue.

We may need it in unique ID validations, but as discussed earlier today, not in deprecated tags scenario

Reply:
I will remove the current manner of dealing with `related issues` (in an upcoming PR) and we will redo this capability in a better manner once we implement the unique ID validation.